src/cache.rs

```diff
@@ -179,16 +224,29 @@ impl Cache {
             })
             .collect();
 
-        // Try to find a platform specific path first.
+        // Get Custom Pages dir
+        let custom_dir = &config.directories.custom_pages_dir;
+
+        // Look up custom page (<name>.page). If it exists, return it directly
+        if let Some(custom_page) = custom_dir.as_ref().map(|dir| dir.join(custom_filename)) {
+            if custom_page.exists() && custom_page.is_file() {
+                return Some(PageLookupResult::with_page(custom_page));
+            }
```

Comment:
You can get rid of the `as_ref` and `.map` if you first unwrap the directory with `if let Some(dir) = custom_dir { ... }` and then call `join` in the nested block.